ssl/ssl_asn1.c

```diff
@@ -356,10 +356,15 @@ SSL_SESSION *d2i_SSL_SESSION(SSL_SESSION **a, const unsigned char **pp,
     ret->ext.max_early_data = as->max_early_data;
 
     if (as->alpn_selected != NULL) {
-        if (!ssl_session_strndup((char **)&ret->ext.alpn_selected,
-                                 as->alpn_selected))
+        if (as->alpn_selected->length > 0) {
+            ret->ext.alpn_selected = OPENSSL_malloc(as->alpn_selected->length);
```

Comment:
ssl_session_strndup opens with OPENSSL_free(*pdst). Now there is no free. Is it safe to assume that alpn_selected is always NULL at this point?

Reply:
Fixed